cmd/argocd/commands/app.go

```diff
@@ -1091,8 +1109,42 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 			conn, appIf := acdClient.NewApplicationClientOrDie()
 			defer util.Close(conn)
 
-			selectedResources := parseSelectedResources(resources)
 			appName := args[0]
+
+			selectedLabels := parseLabels(labels)
+			if len(selectedLabels) > 0 {
```

Comment:
Feels like a good candidate for an e2e test.

Reply:
Added a test!